apps/wallet/src/ui/app/hooks/useUnlockedGuard.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { useAccounts } from '../hooks/useAccounts';
+import { useActiveAccount } from '../hooks/useActiveAccount';
+
+export function useUnlockedGuard() {
+	const navigate = useNavigate();
+	const { data: allAccounts, isLoading: isAccountsLoading } = useAccounts();
+	const activeAccount = useActiveAccount();
+	const loading = isAccountsLoading || !activeAccount;
+	const isInitialized = !!allAccounts?.length;
+	const isLocked = !!activeAccount?.isLocked;
+	const guardAct = !loading && isInitialized && isLocked;
```

Comment:
could we rename `guardAct` to something a bit more clear like `guardAccount` or `isLocked` or `showGuard`

Reply:
The guard will act only when this variable is true. That's why I named it `guardAct`. Not sure what's a better name